### PR TITLE
Recruitment api

### PR DESCRIPTION
**Pull request by HoangCacLoanChau**
## Postman (BE)
- CREATE RECRUITMENT
image
- [ADMIN] GET ALL LIST
image
- [ADMIN] GET BY ID
image
- [ADMIN] APPROVE
image
- [ADMIN] REJECT
image

- [PUBLIC] GET APPROVED AND ARCHIVED IS FALSE LIST
image
- [PUBLIC] GET BY ID AND APPROVED
image

- [OWNER] GET LIST BY OWNER ID
image

- [OWNER] UPDATE
image

-[OWNER] DELETE
image

-[OWNER] GET BY ID
image

## Checklist before requesting a review
1. [ ] I have performed a self-review of my code
2. [ ] Have you lint your code locally before submission?

### CODE REVIEW

File: src/modules/recruitments/services/recruitments.service.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    recruitmentOwnerId: string,
+    body: CreateRecruitmentProps,
+  ) => {
+    const newRecruitment = await tenantPrisma.recruitment.create({
+      data: {
+        ...body,
+        recruitmentOwnerId: recruitmentOwnerId,
+      },
+    });
+    await tenantPrisma.$disconnect();
```
Comment (pvnamGit) [inline src/modules/recruitments/services/recruitments.service.ts:35]: why?

File: src/modules/recruitments/controllers/adminRecruitment.controller.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    res: NextApiResponse<ApiSuccessResponse | ApiErrorResponse>,
+  ) => {
+    try {
+      const prisma = await getPrismaClient(req.tenantId);
+      const { page, limit, companyName, job, salary, position, type } =
+        req.query;
+      const listRecruitment = await AdminRecruitmentService.getList(prisma, {
+        companyName: companyName ? (companyName as string) : '',
+        page: page ? parseInt(page as string, 10) : 1,
+        limit: limit ? parseInt(limit as string, 10) : 20,
```
Comment (pvnamGit) [inline src/modules/recruitments/controllers/adminRecruitment.controller.ts:18]: how about other filters? `job, salary, position, type`

File: src/pages/api/recruitments/admin/[id].tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+const handler = nc({
+  onError: onErrorAPIHandler,
+  onNoMatch: onNoMatchAPIHandler,
+}).use(extractTenantId);
+
+handler
+  .get(verifySchoolAdmin, AdminRecruitmentController.getById)
+  .put(verifySchoolAdmin, AdminRecruitmentController.approve)
+  .delete(verifySchoolAdmin, AdminRecruitmentController.reject);
```
Comment (pvnamGit) [inline src/pages/api/recruitments/admin/[id].tsx:16]: too weird if using `DELETE` method to reject:
I recommend API should be: `.../ap/recruitments/admin?action=approve/reject". And use method `PUT`

File: src/modules/recruitments/services/recruitments.service.ts (hunk 1 above)
Comment (HoangCacLoanChau) [reply src/modules/recruitments/services/recruitments.service.ts:35]: Testing something and I forgot to delete, sorry.

File: src/modules/recruitments/services/recruitments.service.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+export default class RecruimentService {
+  static create = async (
+    tenantPrisma: PrismaClient,
+    recruitmentOwnerId: string,
+    body: CreateRecruitmentProps,
+  ) => {
+    const newRecruitment = await tenantPrisma.recruitment.create({
+      data: {
+        ...body,
+        recruitmentOwnerId: recruitmentOwnerId,
```
Comment (pvnamGit) [inline src/modules/recruitments/services/recruitments.service.ts:32]: also add information of owner

File: prisma/migrations/20230221071305_add_recruitment_owner_info/migration.sql [hunk 5]
```diff
@@ -0,0 +1,6 @@
+-- AlterTable
+ALTER TABLE "recruitments" ADD COLUMN     "recruitment_owner_info_id" TEXT,
+ALTER COLUMN "recruitment_owner_id" DROP NOT NULL;
+
+-- AddForeignKey
+ALTER TABLE "recruitments" ADD CONSTRAINT "recruitments_recruitment_owner_info_id_fkey" FOREIGN KEY ("recruitment_owner_info_id") REFERENCES "informations"("id") ON DELETE SET NULL ON UPDATE CASCADE;
```
Comment (pvnamGit) [inline prisma/migrations/20230221071305_add_recruitment_owner_info/migration.sql:6]: hey hey hey =)))

Comment (HoangCacLoanChau) [reply prisma/migrations/20230221071305_add_recruitment_owner_info/migration.sql:6]: :)) my bab

File: prisma/schema.prisma [hunk 6]
```diff
@@ -241,23 +242,25 @@ model NewsComment {
...
-  expiredAt          DateTime? @map("expired_at")
-  salary             String    @default("negotiate")
-  isApproved         Boolean   @default(false)
-  recruitmentOwnerId String    @map("recruitment_owner_id")
+  id                     String       @id @default(cuid())
+  archived               Boolean      @default(false)
+  createdAt              DateTime     @default(now()) @map("created_at")
+  updatedAt              DateTime     @default(now()) @updatedAt @map("updated_at")
+  companyName            String       @map("company_name")
+  website                String
```
Comment (pvnamGit) [inline prisma/schema.prisma:250]: should be optional, some company does not have a website

File: prisma/schema.prisma [hunk 7]
```diff
@@ -241,23 +242,25 @@ model NewsComment {
...
+  job                    String
+  address                String
+  position               String
+  title                  String
```
Comment (pvnamGit) [inline prisma/schema.prisma:254]: what is the difference between `job` and `title`?

File: prisma/schema.prisma [hunk 8]
```diff
@@ -241,23 +242,25 @@ model NewsComment {
...
+  description            String
+  type                   String       @map("type")
```
Comment (pvnamGit) [inline prisma/schema.prisma:256]: no need to `@map("type")`, it should be optional

File: prisma/schema.prisma (hunk 7 above)
Comment (HoangCacLoanChau) [reply prisma/schema.prisma:254]: image
as you can see, job is your major career, and your title is the name of the recruitment

File: prisma/schema.prisma (hunk 8 above)
Comment (HoangCacLoanChau) [reply prisma/schema.prisma:256]: ok, noted

File: prisma/schema.prisma (hunk 6 above)
Comment (HoangCacLoanChau) [reply prisma/schema.prisma:250]: ok, noted, anything else ?

File: prisma/migrations/20230221114026_drop_not_null_constraint_website_type_description/migration.sql [hunk 9]
```diff
@@ -0,0 +1,5 @@
+SELECT template.run_migration('
+-- AlterTable
+ALTER TABLE "recruitments" ALTER COLUMN "website" DROP NOT NULL,
+ALTER COLUMN "description" DROP NOT NULL;
```
Comment (pvnamGit) [inline prisma/migrations/20230221114026_drop_not_null_constraint_website_type_description/migration.sql:4]: Coi có thiếu cái drop not null của cái type ko đó

Comment (HoangCacLoanChau) [reply prisma/migrations/20230221114026_drop_not_null_constraint_website_type_description/migration.sql]: Prisma bug nha 😀

**Pull request merged**